controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -408,3 +547,60 @@ void wfa_ca::handle_wfa_ca_message(
     }
     }
 }
+
+bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessageType message_type)
+{
+    std::shared_ptr<ieee1905_1::cCmduHeader> cmdu_header;
+
+    int tlv_type;
+
+    switch (ieee1905_1::eMessageType(message_type)) {
+    case ieee1905_1::eMessageType::CHANNEL_PREFERENCE_QUERY_MESSAGE: {
+
+        cmdu_header =
+            cmdu_tx.create(g_mid, ieee1905_1::eMessageType::CHANNEL_PREFERENCE_QUERY_MESSAGE);
+        if (!cmdu_header) {
+            LOG(ERROR) << "Failed building message!";
+            return false;
+        }
+        break;
+    }
+    case ieee1905_1::eMessageType::CHANNEL_SELECTION_REQUEST_MESSAGE: {
+
+        cmdu_header = cmdu_tx.load();
+        if (!cmdu_header) {
+            LOG(ERROR) << "load cmdu has failed";
+            return false;
+        }
+
+        tlv_type = cmdu_tx.getNextTlvType();
+        while (tlv_type != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+            if (tlv_type == int(wfa_map::eTlvTypeMap::TLV_CHANNEL_PREFERENCE)) {
+                if (!cmdu_tx.addClass<wfa_map::tlvChannelPreference>()) {
```

Comment:
Why do you bother parsing this? You don't do anything with the channel preference TLV.

I suppose this is something to be added in a later commit? In that case, I don't see why you wouldn't also delay the parsing until that later commit.

In fact, I don't see the point of this entire function...

Reply:
Because I loaded preprepared buffer containing the data but the CMDU I created is new and empty. So I need to perform AddClass() to every TLV which was added to the buffer, in order to push it to CMDU classes vector.
It is necessary because when we finalize the TLV on the send function,  we iterate over this vector and do swapping.